tools/wptserve/wptserve/server.py

```diff
@@ -369,17 +369,19 @@ def handle_one_request(self):
                             stream_queues[frame.stream_id] = (self.start_stream_thread(frame, queue), queue)
                         stream_queues[frame.stream_id][1].put(frame)
 
-                        if isinstance(frame, StreamEnded) or frame.stream_ended:
+                        if isinstance(frame, StreamEnded) or (hasattr(frame, "stream_ended") and frame.stream_ended):
                             del stream_queues[frame.stream_id]
 
         except (socket.timeout, socket.error) as e:
-            self.logger.debug('(%s) ERROR - Closing Connection - \n%s' % (self.uid, str(e)))
+            self.logger.error('(%s) ERROR - Closing Connection - \n%s' % (self.uid, str(e)))
```

Comment:
The formatter will add `ERROR` to the string in this case, so you don't need to.

Reply:
Ahh yes, brain is elsewhere